controller/konnect/konnectextension_controller.go

```diff
@@ -265,23 +266,44 @@ func (r *KonnectExtensionReconciler) Reconcile(ctx context.Context, req ctrl.Req
 	}
 
 	// Get the GatewayKonnectControlPlane and set conditions accordingly.
+	cpID := lo.FromPtr(ext.Status.Konnect).ControlPlaneID
 	cp, res, err := r.getGatewayKonnectControlPlane(ctx, ext)
 	if err != nil || !res.IsZero() {
-		if !k8serrors.IsNotFound(err) && !errors.Is(err, extensionserrors.ErrKonnectGatewayControlPlaneNotProgrammed) {
+		switch {
+		case !k8serrors.IsNotFound(err) && !errors.Is(err, extensionserrors.ErrKonnectGatewayControlPlaneNotProgrammed):
 			return res, err
+		case k8serrors.IsNotFound(err) && cleanup:
+			log.Debug(logger, "ControlPlane not found, for KonnectExtension during deletion, proceeding with cleanup")
+		default:
+			log.Debug(logger, "ControlPlane not ready yet")
+			return res, nil
 		}
-		log.Debug(logger, "controlPlane not ready yet")
-		return res, nil
+	} else {
+		// Update the controlPlane ID to the value presented in the status by ControlPlane itself.
+		cpID = cp.Status.ID
 	}
 
 	log.Debug(logger, "controlPlane reference validity checked")
 
 	apiAuthRef, err := getKonnectAPIAuthRefNN(ctx, r.Client, &ext)
 	if err != nil {
-		if client.IgnoreNotFound(err) != nil {
+		if k8serrors.IsNotFound(err) {
```

Comment:
Can we reverse the logic here to get rid of unnecessary `else` clause?

e.g. 

```suggestion
		if !k8serrors.IsNotFound(err) {
			return ctrl.Result{}, err
		}
```